File: tests/php/Model/BusinessTest.php
<?php

namespace Trulioo\SDK\Test\Model;

use PHPUnit\Framework\TestCase;
use Trulioo\SDK\Model\Business;

class BusinessTest extends TestCase
{
    private Business $business;

    public function setUp(): void
    {
        $this->business = new Business([]);
    }

    public function testPropertyBusinessName()
    {
        $this->business->setBusinessName('test');
        $this->assertEquals('test', $this->business->getBusinessName());
    }

    public function testPropertyBusinessRegistrationNumber()
    {
        $this->business->setBusinessRegistrationNumber('123');
        $this->assertEquals('123', $this->business->getBusinessRegistrationNumber());
    }

    public function testPropertyDayOfIncorporation()
    {
        $this->business->setDayOfIncorporation(15);
        $this->assertEquals(15, $this->business->getDayOfIncorporation());
    }

    public function testPropertyMonthOfIncorporation()
    {
        $this->business->setMonthOfIncorporation(9);
        $this->assertEquals(9, $this->business->getMonthOfIncorporation());
    }

    public function testPropertyYearOfIncorporation()
    {
        $this->business->setYearOfIncorporation(2020);
        $this->assertEquals(2020, $this->business->getYearOfIncorporation());
    }

    public function testPropertyJurisdictionOfIncorporation()
    {
        $this->business->setJurisdictionOfIncorporation('Alberta');
        $this->assertEquals('Alberta', $this->business->getJurisdictionOfIncorporation());
    }

    public function testPropertyShareholderListDocument()
    {
        $this->business->setShareholderListDocument(true);
        $this->assertEquals(true, $this->business->getShareholderListDocument());
    }

    public function testPropertyFinancialInformationDocument()
    {
        $this->business->setFinancialInformationDocument(false);
        $this->assertEquals(false, $this->business->getFinancialInformationDocument());
    }

    public function testPropertyDunsNumber()
    {
        $this->business->setDunsNumber('123');
        $this->assertEquals('123', $this->business->getDunsNumber());
    }

    public function testPropertyEntities()
    {
        $this->business->setEntities(true);
        $this->assertEquals(true, $this->business->getEntities());
    }
}

File: tests/unit/java/com/trulioo/sdk/model/BusinessTest.java
package com.trulioo.sdk.model;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.*;

public class BusinessTest {
    private Business business;

    @BeforeEach
    public void beforeEach() {
        business = new Business();
    }

    @Test
    public void testBusinessName() {
        String businessName = "test";
        business.setBusinessName(businessName);

        assertEquals(businessName, business.getBusinessName());
        assertEquals(businessName, new Business().businessName(businessName).getBusinessName());
    }

    @Test
    public void testBusinessRegistrationNumber() {
        String businessRegistrationNumber = "123";
        business.setBusinessRegistrationNumber(businessRegistrationNumber);

        assertEquals(businessRegistrationNumber, business.getBusinessRegistrationNumber());
        assertEquals(businessRegistrationNumber,
                new Business().businessRegistrationNumber(businessRegistrationNumber).getBusinessRegistrationNumber());
    }

    @Test
    public void testDayOfIncorporation() {
        Integer dayOfIncorporation = 15;
        business.setDayOfIncorporation(dayOfIncorporation);

        assertEquals(dayOfIncorporation, business.getDayOfIncorporation());
        assertEquals(dayOfIncorporation, new Business().dayOfIncorporation(dayOfIncorporation).getDayOfIncorporation());
    }

    @Test
    public void testMonthOfIncorporation() {
        Integer monthOfIncorporation = 9;
        business.setMonthOfIncorporation(monthOfIncorporation);

        assertEquals(monthOfIncorporation, business.getMonthOfIncorporation());
        assertEquals(monthOfIncorporation,
                new Business().monthOfIncorporation(monthOfIncorporation).getMonthOfIncorporation());
    }

    @Test
    public void testYearOfIncorporation() {
        Integer yearOfIncorporation = 2020;
        business.setYearOfIncorporation(yearOfIncorporation);

        assertEquals(yearOfIncorporation, business.getYearOfIncorporation());
        assertEquals(yearOfIncorporation,
                new Business().yearOfIncorporation(yearOfIncorporation).getYearOfIncorporation());
    }

    @Test
    public void testJurisdictionOfIncorporation() {
        String jurisdictionOfIncorporation = "Alberta";
        business.setJurisdictionOfIncorporation(jurisdictionOfIncorporation);

        assertEquals(jurisdictionOfIncorporation, business.getJurisdictionOfIncorporation());
        assertEquals(jurisdictionOfIncorporation, new Business()
                .jurisdictionOfIncorporation(jurisdictionOfIncorporation).getJurisdictionOfIncorporation());
    }

    @Test
    public void testShareholderListDocument() {
        Boolean shareholderListDocument = true;
        business.setShareholderListDocument(shareholderListDocument);

        assertEquals(shareholderListDocument, business.getShareholderListDocument());
        assertEquals(shareholderListDocument,
                new Business().shareholderListDocument(shareholderListDocument).getShareholderListDocument());
    }

    @Test
    public void testFinancialInformationDocument() {
        Boolean financialInformationDocument = false;
        business.setFinancialInformationDocument(financialInformationDocument);

        assertEquals(financialInformationDocument, business.getFinancialInformationDocument());
        assertEquals(financialInformationDocument, new Business()
                .financialInformationDocument(financialInformationDocument).getFinancialInformationDocument());
    }

    @Test
    public void testDunsNumber() {
        String dunsNumber = "123";
        business.setDunsNumber(dunsNumber);

        assertEquals(dunsNumber, business.getDunsNumber());
        assertEquals(dunsNumber, new Business().dunsNumber(dunsNumber).getDunsNumber());
    }

    @Test
    public void testEntities() {
        Boolean entities = true;
        business.setEntities(entities);

        assertEquals(entities, business.getEntities());
        assertEquals(entities, new Business().entities(entities).getEntities());
    }

    @Test
    public void testEquals() {
        String businessName = "test";
        Business business1 = new Business().businessName(businessName);

        assertEquals(business1, business1);
        assertEquals(business1, new Business().businessName(businessName));
        assertNotEquals(business1, new Business().businessName(businessName + "1"));
        assertNotEquals(business1, null);
    }

    @Test
    public void testHashcode() {
        String businessName = "test";

        assertEquals(new Business().businessName(businessName).hashCode(),
                new Business().businessName(businessName).hashCode());
    }

    @Test
    public void testToString() {
        String businessName = "test";

        assertEquals(new Business().businessName(businessName).toString(),
                new Business().businessName(businessName).toString());
    }
}
